test/RewardPool/APR.test.ts

```diff
@@ -0,0 +1,155 @@
+      const managerRole = await rewardPool.MANAGER_ROLE();
+
+      expect(await rewardPool.hasRole(managerRole, this.signers.system.address)).to.be.true;
+      expect(await rewardPool.base()).to.be.equal(INITIAL_BASE_APR);
+      expect(await rewardPool.macroFactor()).to.be.equal(INITIAL_MACRO_FACTOR);
+      expect(await rewardPool.rsi()).to.be.equal(INITIAL_RSI_BONUS);
+      expect(await rewardPool.EPOCHS_YEAR()).to.be.equal(EPOCHS_YEAR);
+      expect(await rewardPool.DENOMINATOR()).to.be.equal(DENOMINATOR);
+    });
+
+    it("should initialize vesting bonus", async function () {
+      const { rewardPool } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+      expect(await rewardPool.getVestingBonus(1)).to.be.equal(6);
+      expect(await rewardPool.getVestingBonus(52)).to.be.equal(2178);
+    });
+
+    it("should get initial RSI", async function () {
+      const { rewardPool } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+      expect(await rewardPool.getDefaultRSI()).to.be.equal(INITIAL_RSI_BONUS);
+    });
+
+    it("should get max RSI", async function () {
+      const { rewardPool } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+      expect(await rewardPool.getMaxRSI()).to.be.equal(MAX_RSI);
+    });
+
+    it("should get max APR", async function () {
+      const { rewardPool } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+      const base = await rewardPool.base();
+      const macroFactor = await rewardPool.macroFactor();
+      const vestingBonus = await rewardPool.getVestingBonus(52);
+      const rsiBonus = await rewardPool.getMaxRSI();
+      const nominator = base.add(vestingBonus).mul(macroFactor).mul(rsiBonus);
+      const denominator = DENOMINATOR.mul(DENOMINATOR).mul(DENOMINATOR);
+
+      const maxAPR = await rewardPool.getMaxAPR();
+      expect(maxAPR.nominator).to.be.equal(nominator);
+      expect(maxAPR.denominator).to.be.equal(denominator);
+    });
+
+    it("should apply max rewards", async function () {
+      const { rewardPool } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+      const reward = ethers.BigNumber.from(10);
+      const maxReward = await applyMaxReward(rewardPool, reward);
+
+      expect(await rewardPool.applyMaxReward(reward)).to.be.equal(maxReward);
+    });
+
+    it("should get epoch max reward", async function () {
+      const { rewardPool } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+      const totalStaked = ethers.BigNumber.from(1000);
+      const maxAPR = await rewardPool.getMaxAPR();
+      const maxEpochReward = totalStaked.mul(maxAPR.nominator).div(maxAPR.denominator).div(EPOCHS_YEAR);
+
+      expect(await rewardPool.getEpochMaxReward(totalStaked)).to.be.equal(maxEpochReward);
+    });
+  });
+
+  describe("Setters", function () {
```

Comment:
I prefer separating the tests with describe based on business logic (like describe("RSI"), describe(MacroFactor), etc. instead of "Setters")